aiosmtpd/smtp.py

```diff
@@ -239,36 +305,29 @@ async def _handle_client(self):
         while self.transport is not None:   # pragma: nobranch
             # XXX Put the line limit stuff into the StreamReader?
             try:
-                line = await self._reader.readline()
+                line: bytes = await self._reader.readline()
                 log.debug('_handle_client readline: %s', line)
                 # XXX this rstrip may not completely preserve old behavior.
                 line = line.rstrip(b'\r\n')
                 log.info('%r Data: %s', self.session.peer, line)
                 if not line:
                     await self.push('500 Error: bad syntax')
                     continue
-                i = line.find(b' ')
+                command, _, arg = line.partition(b" ")
```

Comment:
`str.partition()` and `bytes.partition()` are amazingly fast and result in simpler follow-up code. Let's use them more :-)

Edit: The reason I replaced this was actually an attempt to make `_handle_client` faster. Because in PyPy3 all these for unknown reasons run slower, and enabling PYTHONASYNCIODEBUG flag points me that this method is one of the main reasons for the slowness. This speeds up `_handle_client` enough to greatly reduce the number of "slow callback" warning when testing against PyPy3